www/annotationdb/annotationdb.js

```diff
@@ -0,0 +1,382 @@
+			$(this).treeview('selectNode', [ new_selections, {silent: true} ]);
+
+			generate_query();
+		},
+		onNodeUnselected: function(event, node){
+
+			var unselections = select_nodes(node, []);
+			var tree = $(this).treeview(true);
+
+			$(this).treeview('selectNode', [ unselections, {silent: true} ]);
+			$(this).treeview('unselectNode', [ unselections, {silent: true }]);
+
+			generate_query();
+		}
+	});
+}
+
+function build_table(node, id) {
+
+	var table_id = id + '-table';
+
+	$('#' + id + '>.panel-body').append(
+		'<table id="' + table_id + '" class="table table-hover" data-detail-view="true">' +
+			'<thead>' + 
+				'<th class="col0 detail"></th>' +
+				'<th class="col1">Variable</th>' +
+				'<th class="col2">Type</th>' +
+				'<th class="col3">Description</th>' +
+			'</thead>' +
+			'<tbody>' +
+			'</tbody>' +
+		'</table>'
+	);
+
+	$.each(node['nodes'], function(_, child_node) {
+
+		var col0 = '';
+		var col1 = child_node['annotation'];
+		var col2 = child_node['type'];
+		var col3 = child_node['description'];
+
+		var has_elements = (child_node.hasOwnProperty('elements') ? true : false);
+
+		if (has_elements) {
+
+			var sub_table_id = 'sub-' + table_id + '-' + child_node['annotation'].split('.').slice(-1)[0];
+
+			col0 = '<a data-toggle="collapse" href="#row-' + sub_table_id + '" aria-expanded="false" aria-controls="row-' + sub_table_id + '">' +
+				       '<i class="sub-table-icon glyphicon glyphicon-plus icon-plus"><i>' +
+				   '</a>';
+
+		}
+
+		$('#' + table_id + '>tbody').append(
+			'<tr>' +
+				'<td class="col0">' + col0 + '</td>' +
+				'<td class="col1"><span class="hail-code">' + col1 + '</span></td>' +
+				'<td class="col2"><span class="hail-code">' + col2 + '</span></td>' +
+				'<td class="col3"><span>' + col3 + '</span></td>' +
+			'</tr>'
+		);
+
+		if (has_elements) {
+
+			$('#' + table_id + '>tbody').append(
+				'<tr id="row-' + sub_table_id + '" class="collapse">' +
+					'<td colspan="4" class="sub-table">' +
+						'<table id="' + sub_table_id + '" class="table table-hover sub-table">' + 
+							'<thead>' +
+								'<th>Struct Element</th>' +
+								'<th>Type</th>' +
+							'</thead>' +
+							'<tbody>' +
+							'</tbody>' +
+						'</table>' +								
+					'</td>' +
+				'</tr>'
+			);
+
+			$.each(child_node['elements'], function(_, element){
+				$('#' + sub_table_id + '>tbody').append(
+					'<tr>' +
+						'<td class="sub-col0"><span class="hail-code">' + element['name'] + '</span></td>' +
+						'<td class="sub-col1"><span class="hail-code">' + element['type'] + '</span></td>' +
+					'</tr>'
+				);
+			});
+		}
+
+	});
+	
+}
+
+function build_docs(nodes, parent_id, parent_level) {
+
+	var level = parent_level + 1;
+	var current_parent = parent_id;
+
+	$.each(nodes, function(_, node) {
+
+		var node_id = current_parent + '-' + node['annotation'].split('.').slice(-1)[0];
+
+		$('#' + current_parent).append(
+			'<div class="panel panel-default level' + String(level) + '">' +
```

Comment:
Would it be difficult to add a plus/minus symbol on the right corner of each panel? I think it would be clearer that they expand if there was an indicator.